NCRC Application and HPC End User Documentation Improvements

milljm opened the pull request:
This is a work in progress change for NCRC How-To

Closes #22916 #23190

Commit:
WIP NCRC Application End User

This is a work in progress change for NCRC How-To

Closes #22916

Commit:
Fix template variable typo

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1260197/) on 8da5dd1 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/22922/site)

This comment will be updated on new commits.

Commit:
Fix Bison's source access link

Commit:
Requested changes

Add SAM, RELAP5
Add Sabertooth, Dire Wolf, and Blue Crab descriptions.

Commit:
Template cluster information

Commit:
Add RELAP5 Description

Commit:
Add binary_method variable

Some Applications don't have a method suffix (relap5). In order to continue
using the templates, I need to separate {{binary}} and {{binary_method}}.

Commit:
Add RELAP5 Documentation link

Commit:
Update SAM's description

Commit:
bah...

Commit:
adding small logos (update large_media)

moosebuild commented on large_media:
```diff
@@ -1 +1 @@
-Subproject commit 8d84cd9f3cffe24b781a005acb4c149767fcb56c
+Subproject commit 05a5757c5a1806bc829c3c651b87dba1ad1981a9
```
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

Commit:
Trying to fix mobile CSS issues

Commit:
WIP CSS changes

Commit:
WIP Changes to CSS for Mobile devices

Commit:
Changes reflecting github.inl.gov transition

Changes to support the transition to github.inl.gov.

More templates for source code access.

WIP: where necessary, depict some instructions as advanced, while
instead deferring to HPC OnDemand (which requires no SSH config
changes, and _just works_)

Closes #22916 #23190

moosebuild commented on large_media:
```diff
@@ -1 +1 @@
-Subproject commit 2aced06ad10236c57b4773102defc0b905020249
+Subproject commit 05a5757c5a1806bc829c3c651b87dba1ad1981a9
```
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

milljm commented on modules/doc/config.yml:
```diff
@@ -44,7 +44,7 @@ Extensions:
                 Tutorials: getting_started/examples_and_tutorials/index.md
                 Supplemental Resources: getting_started/supplemental_resources/index.md
-                INL HPC Remote Access: help/inl/index.md
+                INL HPC Services: help/inl/index.md
```
"Remote Access" no longer encompasses what we are doing in this section

cticenhour replied:
I think this is a good change that reflects how broad this page now is - resolving this comment.

milljm commented on modules/doc/content/help/inl/index.md:
```diff
@@ -10,7 +10,7 @@ available codes and how to utilize the [!ac](INL) resources to run applications.
 
 
-## Requesting Access id=access
+# Requesting Access id=access
```
Git rid of the right table of contents bar. We don't need it for one item...

cticenhour replied:
👍🏻

Commit:
Looking good...

Templated the Level 1 and 4 instructions to personalize the users
experience, tailored to the Application they wish to use.

Trimmed down the landing page. We want to 'herd' users to the
application page as much as possible. Where each app is listed with
each level of access, etc.

- NCRC Distributed Applications
- HPC OnDemand
- Remote Access Primer

cticenhour commented on modules/doc/content/ncrc/applications/hpc_application_template.md:
```diff
@@ -133,42 +85,52 @@ module load use.moose moose-apps {{ApplicationLower}}
-to copy the output of this test run use the following command on your local machine.
+!alert note title=Copying files from remote HPC machine to your machine
+Copying files from an HPC cluster to your machine first requires that you follow the instructions:
+[inl/hpc_remote.md]. It is for advanced users, comfortable with modifying their SSH settings, and
+being familiar with their terminal in general.
```
```suggestion
[inl/hpc_remote.md]. This workflow is for advanced users, comfortable with modifying their SSH settings, and
familiar with their terminal in general.
```

cticenhour commented on modules/doc/content/ncrc/applications/hpc_application_template.md:
```diff
@@ -177,7 +139,7 @@ In general, scheduling jobs requires two steps:
 ### (1) Create [!ac](PBS) Script
 
-Using the example problem from above a simple script for scheduling may be created. The top
+Lets use the same example input file from above to create a simple script for scheduling. The top
```
```suggestion
Let's use the same example input file from above to create a simple script for scheduling. The top
```

cticenhour commented on modules/doc/content/ncrc/applications/ncrc_level4_template.md:
```diff
@@ -0,0 +1,49 @@
+# {{ApplicationName}} Source
+
+!alert tip title=Existing {{ApplicationName}} HPC GitLab Users
+Existing users using [hpcgitlab.hpc.inl.gov](https://hpcgitlab.hpc.inl.gov), please see the [Transitional Guide](https://github.com/idaholab/moose/wiki/NCRC-github.inl.gov-transition-guide),
+for help on accessing the new home for [NCRC/{{ApplicationName}}](https://github.inl.gov/ncrc/{{binary}}). New users can safely ignore this notice.
```
```suggestion
For existing users who access code via [hpcgitlab.hpc.inl.gov](https://hpcgitlab.hpc.inl.gov), please see the [Transitional Guide](https://github.com/idaholab/moose/wiki/NCRC-github.inl.gov-transition-guide)
for help on accessing the new home for [NCRC/{{ApplicationName}}](https://github.inl.gov/ncrc/{{binary}}). New users can safely ignore this notice.
```

cticenhour commented on modules/doc/content/ncrc/applications/ncrc_level4_template.md:
```diff
@@ -0,0 +1,49 @@
+
+Click 'Select an owner' and choose yourself. Take note of how GitHub identifies you
+(`first-last` name). Then click 'Create fork' near the bottom. GitHub will begin creating a
+personalized fork of {{ApplicationName}}. When finished GitHub will bring you to your fork of
```
```suggestion
personalized fork of {{ApplicationName}}. When finished, GitHub will bring you to your fork of
```

cticenhour commented on modules/doc/content/ncrc/applications/ncrc_ondemand_level4.md:
```diff
@@ -0,0 +1,83 @@
+## Submission and Job Information
+
```
Lots of extra blank lines in this doco - consider removing some of them.

milljm replied:
nice! another file we don't need. It too was converted into template `ncrc_ondemand_template.md`. I think I created this file by mistake. But I will fix the excessive lines in the renamed template.

milljm commented on modules/doc/content/help/inl/hpc_binary.md:
````diff
@@ -3,45 +3,16 @@
-   ```
+Familiarize yourself with the interactive shell available with [inl/hpc_ondemand.md], and return
+here with such a shell on the machine you wish to run your application. Most users will want to
+operate on Sawtooth, although the following site details all available clusters for use
````
Good find. I need to delete this file all together. It was converted into a template `ncrc_application_template.md`

Commit:
Addressed comments (acronyms, spelling)

Added NCRC/NEAMS acronyms.
Deleted no longer used files.

milljm commented on modules/doc/content/help/inl/applications.md:
```diff
@@ -1,16 +1,22 @@
-focused on nuclear energy, but as MOOSE and [!ac](INL) simulation capabilities continue to grow the
-breadth of applications continues to expand as well. The following sections highlight some of the
-[!ac](INL) developed applications.
+NCRC distributes a number of NEAMS-developed applications. The following sections highlight some of the codes available through the NCRC.
```
👍

cticenhour commented on framework/doc/acronyms.yml:
```diff
@@ -34,6 +34,8 @@ MWR: Method of Mean Weighted Residuals
 MPI: Message Passing Interface
 MOOSE: Multiphysics Object Oriented Simulation Environment
 NE: Nuclear Energy
+NEAMS: Nuclear Engineering Advanced Modeling and Simulation
```
```suggestion
NEAMS: Nuclear Energy Advanced Modeling and Simulation
```

My mistake here - should be "Energy" instead.

milljm replied:
👍

Commit:
Fix NEAMS acronym

Commit:
Use application name instead of 'the binary'

Commit:
Need to specify a development directory

Other instructions are expecting ~/projects, so lets add that here.

Commit:
This site should exactly match inl/index

Rather than delete this, lets include the real site. I believe
management may still be including the URL which lands here in
emails to end-users.

Commit:
More files no longer sourced/used

Commit:
Create root level entry points to apps

The external app's 'Getting Started' pages can benifit from these
root_app markdowns as their includes.

milljm commented:
MOAR TEMPLATES

cticenhour commented on modules/doc/content/ncrc/applications/ncrc_level4_template.md:
```diff
@@ -0,0 +1,69 @@
+For existing users who access code via [hpcgitlab.hpc.inl.gov](https://hpcgitlab.hpc.inl.gov),
+please see the [Transitional Guide](https://github.com/idaholab/moose/wiki/NCRC-github.inl.gov-transition-guide)
+for help on accessing the new home for [NCRC/{{ApplicationName}}](https://github.inl.gov/ncrc/{{binary}}). New users can safely ignore this notice.
+
```
```suggestion
```

Remove this extra blank line

cticenhour commented on modules/doc/content/ncrc/applications/ncrc_level4_template.md:
```diff
@@ -0,0 +1,69 @@
+
+Entering a passphrase (or a blank response) is your preference. You'll need to enter this passphrase
+each time you perform a push/pull operation with your fork.
+
```
```suggestion
```

cticenhour commented on modules/doc/content/ncrc/applications/root_marmot.md:
```diff
@@ -0,0 +1,19 @@
+
+!row!
+!col small=12 medium=8 large=8
+MARMOT is a mesoscale fuel performance code. As such, it can predict the evolution of the microstructure and material properties of fuels and claddings due to stress, temperature, and irradiation damage. MARMOT can, therefore, supply microstructure-based materials models to other applications that work at a larger scale, such as BISON which works at an engineering scale. MARMOT solves equations i ...
```
```suggestion
MARMOT is a mesoscale fuel performance code. As such, it can predict the evolution of the microstructure and material properties of fuels and claddings due to stress, temperature, and irradiation damage. MARMOT can, therefore, supply microstructure-based materials models to other applications that work at a larger scale, such as BISON at the engineering scale. MARMOT solves equations involving solid mechanics and heat conduction using the finite element method.
```

cticenhour reviewed (commented):
A few more things

Commit:
Adding Development template

Adding make/runtest templates to help generalize that aspect of
MOOSE based application development.

Filenames were becoming an issue. Renamed everything NCRC related
to an ncrc prefix theme (sorry all).

Commit:
grammar/syntax corrections

Commit:
More grammar/syntax fixes

Commit:
remove extra lines

Commit:
correct spelling mistake

joshuahansel commented on modules/doc/content/ncrc/applications/ncrc_develop.md.template:
```diff
@@ -0,0 +1,159 @@
+`git fetch upstream` updates your local references with that of the remote's. It does not 'modify'
+any source files. It will return nothing if there was nothing new to fetch on the `upstream` remote.
+
+`git branch` reports what branch you have checked out. It is impossible to write a document for
+every application's branch naming schemes, so whatever was reported with `git branch`, use that for
+the next command.
```
This probably needs to be clarified. `git branch` will in general return multiple branch names (remember that this section is also read for people *updating* the app). I'd delete "git branch" from your steps above and instead give an explanation of which branch to use:

```
git rebase -i upstream/somebranchname
```
where `somebranchname` is the default branch of {{ApplicationName}}, which can be found using
```
git remote show upstream | grep "HEAD branch" | sed 's/.*: //'
```

milljm replied:
Yeah, I was wrong about this... The correct one-liner is:
```bash
git rev-parse --abbrev-ref HEAD
```

joshuahansel commented on modules/doc/content/ncrc/applications/ncrc_develop.md.template:
```diff
@@ -0,0 +1,159 @@
+If there were errors during the rebase process, that means you have changes which either need
+committing, or conflicts which need addressing.
+
+The {{binary}} repository is now up to date, and understands how to update it's submodules. Do that
```
No apostrophe in "its" for possessive, just the contraction "it is".

joshuahansel commented on modules/doc/content/ncrc/applications/ncrc_develop.md.template:
````diff
@@ -0,0 +1,159 @@
+git submodule update --init
+```
+
+If you whitness any updates to `{{binary}}/moose`, now is the time to apply those two earlier
+warnings about also needing to update your Conda packages or rebuilding PETSc and libMesh:
````
- whitness -> witness. But "observe" is probably a better word here.
- Earlier in your warnings, you made it sound like you should always update the other, but here, there is a conditional (if you witness). You should probably make this consistent - probably amend the earlier warning.

milljm replied:
I realize there are times an update or rebuild is unnecessary, but I am trying to sear this notion into their development process;

- If you update your submodules
  - you should update conda
- if you update conda packages
  - your should update your submodules  

I realize this isn't always 100% true. But if they do it anyway, they will never have an issue... so 🤷 why not.

joshuahansel replied:
That's fine, just wanted to make them consistent, which you have 👍

joshuahansel commented on modules/doc/content/ncrc/applications/ncrc_develop.md.template:
````diff
@@ -0,0 +1,159 @@
+
+```bash
+cd ~/projects/{{binary}}
+make clobberall
````
In the past, there has been concern about `make clobberall` not being enough. Personally, I'm an advocate for "git clean -Xfd".

joshuahansel replied:
Note: I don't know if `make clobberall` also cleans out the submodules. For `git clean`, I essentially do:
```
git clean -Xfd
git submodule foreach --recursive git clean -Xfd
```

milljm replied:
`make clobberall` is the right method. It follows MOOSE_DIR, which may not exist as that applications submodule.

milljm commented on modules/doc/content/ncrc/applications/ncrc_develop.md.template:
```diff
@@ -0,0 +1,159 @@
+before continuing. Please see our [Troubleshooting](help/troubleshooting.md) guide, which also
+contains information on how to reach us.
+
+## Compiler Stack id=compiler
```
Sorry, but I must refuse this request.

The TOC on the right cuts off lengthy text. This is something I've specifically addressed while building these instructions. Your suggesting will end up creating:

- Setting Up the...
- Cloning Sockeye
- Getting the Sup...
- Building Sockeye
- Running Tests

Commit:
reorder the build instructions

Reorder the build instructions to eliminate the double build
of PETSc and libMesh.

Address some concerns with grammar.

milljm commented on modules/doc/content/ncrc/applications/ncrc_develop.md.template:
````diff
@@ -0,0 +1,159 @@
+
+```bash
+cd ~/projects/{{binary}}
+git fetch upstream
````
It was, during the Clone Step.

joshuahansel commented on modules/doc/content/ncrc/applications/ncrc_develop.md.template:
```diff
@@ -0,0 +1,159 @@
+`Then you should perform a Conda update` (if you are using Conda).
+
+!alert warning title=You updated Conda?
+`Then you should perform a submodule update`.
```
I'm guessing you still prefer the two boxes to one? My main issue is that the warning title should be a brief one-line summary of the warning, not a question. You could do "Submodule update needed every Conda update" for example.

joshuahansel replied:
Also, for the second box, you should add some kind of clause that says "if you're using the Conda compiler stack" (doesn't need to be in title, just body)

Commit:
Improve headings to sections

Match section names with the wordage used in the main topic

joshuahansel commented on modules/doc/content/ncrc/applications/ncrc_develop.md.template:
```diff
@@ -0,0 +1,159 @@
+
+  !alert tip title=MOOSE_JOBS? METHODS?
+  Go to [MOOSE Build System](application_development/build_system.md) for more information on
+  available methods. MOOSE_JOBS controls how many cores to use during the use of our scripts.
```
Do you object to this change? My issues are
- I don't see it as a "tip" - I think it's important info to clarify your previous text
- Asking questions in the box title

joshuahansel commented on modules/doc/config.yml:
```diff
@@ -45,7 +45,7 @@ Extensions:
                 Tutorials: getting_started/examples_and_tutorials/index.md
                 Supplemental Resources: getting_started/supplemental_resources/index.md
-                INL HPC Remote Access: help/inl/index.md
+                INL HPC Services: help/inl/index.md
```
I still recommend this one.

joshuahansel commented on modules/doc/content/ncrc/applications/ncrc_develop.md.template:
```diff
@@ -0,0 +1,147 @@
+If this is your first time following these instructions there should be no updates. However, we will
+perform an update to familiarize yourself with the entire process.
+
+{{ApplicationName}} developers will expect users to have performed the following sections, whenever
```
I think this sentence can be deleted. If you want to emphasize that this WHOLE section needs to be completed to update, maybe you can make an `!alert note` after this box with your sentence "Note this step does not actually modify...".

joshuahansel commented on modules/doc/content/ncrc/applications/ncrc_develop.md.template:
````diff
@@ -0,0 +1,147 @@
+Next, we will perform a `rebase` against the provided upstream branch name:
+
+```bash
+git rebase upstream/branch   # replace 'branch' with the discovered branch name
````
I think put your comment in the text instead. I bet some people will copy and paste it...

milljm replied:
Yeah I can make that text instead.

joshuahansel commented on modules/doc/content/ncrc/applications/ncrc_develop.md.template:
````diff
@@ -0,0 +1,147 @@
+
+  ```bash
+  module load use.moose moose-dev    # a reminder to have these loaded
+  export MOOSE_JOBS=6 METHODS=opt    # a reminder to set these
````
I'd probably delete these comments too. I think any time it's a console command rather than lines of a script, you should avoid comments.

milljm replied:
The reason these are here, is that users all the time forget to have these loaded. Logan and I then receive an email forwarded from the NCRC guys, from the customer... and this was the issue.

milljm replied:
Removed these comments. Its harmless to repeat the commands if they previously ran them.

joshuahansel commented on modules/doc/content/ncrc/applications/ncrc_develop.md.template:
````diff
@@ -0,0 +1,147 @@
+```
+
+`make clobberall` deletes the stale object/library files left behind from a previous build. While not
+always necessary, please get into the habit of performing this operation before running `make`.
````
I think this is a little too strict. Maybe add an `!alert note` suggesting that this should be done as a troubleshooting step whenever a build fails and mention why.

milljm replied:
I can make this an alert instead.

What makes sense to you a few months later, when they are trying to perform an update?

This instead?
```bash
git clean -xfd
git submodule foreach --recusive git clean -xfd
```

milljm replied:
Added an alert annotation on running `make clobberall` if the make fails (and if they are trying to perform an update)

Commit:
Relink level-4 with development pages

Move the content from Application Development into what is the
Level - 4 (Source Access) landing page. It makes more sense to
have that content first, and then link later, how to obtain the
source.

Rename the sub sections in Build section, to match what was being
described earlier in the initial main heading.

Other syntax/grammar changes.

milljm commented on modules/doc/content/help/inl/index.md:
```diff
@@ -1,16 +1,14 @@
-# INL HPC Remote Access
+# INL HPC Services
```
This is how folks gain an account within the HPC, which might be for reasons _not_ NCRC related. Though they are still being redirected to NCRC.

Ref:
https://hpc.inl.gov/SitePages/Home.aspx


Yeah, we've had lengthy discussions on what to name this page. I thought we fixed it with: https://mooseframework.inl.gov/ncrc/... so maybe this was a regression. I'll fix it.

milljm replied:
With the newly added NCRC Application drop down, I'd like to keep 'INL HPC Services', as there are still links for HPC only resources (HPC OnDemand, requesting an account)

Commit:
Adresses NCRC/INL/NEAMS comments

Commit:
Move the landing page, grammar, syntax, etc

Move the NCRC content to the NCRC Application page.

Miscellaneous grammar/syntax changes.

Mention recursive submodules as a possible failure point and how
to solve it.

loganharbour committed:
Add more detail to HPC landing page

loganharbour committed:
Remove application logos

They format horribly

loganharbour committed:
Rename discussion forum to support forum

loganharbour committed:
Remove discourse link from apps that don't have it

loganharbour committed:
Correct blue_crab doc link

loganharbour committed:
Remove latest in doc urls

loganharbour committed:
Minor changes to develop template

loganharbour committed:
Changes to level 4 clone

loganharbour committed:
Minor cleanup to apps

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1318826/) on be7ffeb wanted to post the following:

The following commit(s) contain fixups:

- 7628f25b420: fixup! 81f25059c035bbfcb22e5c4d7776641648387e8e

loganharbour committed:
Add login instructions, make key generation more detailed

cticenhour commented on modules/doc/content/ncrc/applications/ncrc_ondemand.md.template:
````diff
@@ -1,68 +1,58 @@
+```
 
- You are allowed to put more than one in this box. Everything that you put in this box will be entered between  `app-opt` and your input file `-i /home/user/input.i`.
+You are allowed to put more than one in this box. Everything that you put in this box will be entered between  `{{binary_method}}` and your input file `-i /home/user/input.i`.
````
```suggestion
You are allowed to put more than one in this box. Everything that you put there will be entered between  `{{binary_method}}` and your input file `-i /home/user/input.i`.
```

cticenhour commented on modules/doc/content/ncrc/applications/ncrc_level4.md.template:
```diff
@@ -0,0 +1,115 @@
+The preferred method for cloning (pulling onto your local machine) repositories is via SSH.
+With this, we will describe how to add SSH keys to [github.inl.gov](https://github.inl.gov)
+that will serve as the authentication method for cloning repsitories. This process only
+needs to be done once per device, that is, you do not need to do it for each NCRC application
```
```suggestion
needs to be done once per device; that is, you do not need to do it for each NCRC application
```

cticenhour commented on modules/doc/content/ncrc/applications/ncrc_level4.md.template:
````diff
@@ -0,0 +1,115 @@
+cat ~/.ssh/id_rsa.pub
+```
+
+and copy the contents of said file. +Only+  copy the results of the command, do not include
````
```suggestion
and copy the contents of said file. +Only+ copy the results of the command, do not include
```

cticenhour commented on modules/doc/content/ncrc/applications/ncrc_root_relap7.md:
```diff
@@ -0,0 +1,16 @@
+
+!row!
+!col small=12 medium=8 large=8
+RELAP-7 is a two-phase thermal systems code based on MOOSE's thermal hydraulics module. RELAP-7 provides two-phase components suitable for LWRs provides closures for two-phase water from TRACE.
```
```suggestion
RELAP-7 is a two-phase thermal systems code based on the MOOSE thermal hydraulics module. RELAP-7 provides two-phase components suitable for LWRs provides closures for two-phase water from TRACE.
```

cticenhour commented on modules/doc/content/help/inl/index.md:
```diff
@@ -1,34 +1,25 @@
 
-INL applications may be obtained by outside institutions and individuals for execution and/or
-development. Currently, the following access methods are available for certain applications.
+HPC ONDemand is a convenient way to access HPC resources (files, submitting jobs, virtual sessions, etc)
```
```suggestion
HPC OnDemand is a convenient way to access HPC resources (files, submitting jobs, virtual sessions, etc)
```

loganharbour committed:
Address reviewer's comments

Co-authored-by: Casey Icenhour <cticenho@ncsu.edu>

cticenhour reviewed (dismissed):
This looks good to me. @joshuahansel do you have any more comments?

loganharbour commented:
Good to go soon. Just waiting on bison + sockeye fixes.

joshuahansel commented on modules/doc/content/ncrc/applications/ncrc_conda_relap7.md:
```diff
@@ -0,0 +1 @@
+!template load file=ncrc/applications/ncrc_conda.md.template ApplicationName=Relap-7 ApplicationLower=relap7 binary=relap-7 binary_method=relap-7-opt
```
```suggestion
!template load file=ncrc/applications/ncrc_conda.md.template ApplicationName=RELAP-7 ApplicationLower=relap7 binary=relap-7 binary_method=relap-7-opt
```

loganharbour replied:
No

joshuahansel replied:
Why not?

joshuahansel commented:
Just the RELAP-7 change above for me.

loganharbour committed:
Correct relap app name

Co-authored-by: joshuahansel <joshua.hansel@inl.gov>

cticenhour reviewed (approved):
Thanks for your input @joshuahansel!

milljm commented:
Need to update information on using HPC Binary: moose (combined-opt)

loganharbour commented:
Please do it in another PR. I'm almost done updating apps and I really need to get these changes in for the NCRC e-mail

cticenhour commented on modules/doc/content/ncrc/applications/ncrc_develop.md.template:
```diff
@@ -0,0 +1,154 @@
+instructions aim at preparing your machine or environment for such development.
+
+If you encounter errors during any of these steps, you need to stop and figure out a solution before
+continuing. Please see our [Troubleshooting](help/troubleshooting.md) guide, which also contains
```
```suggestion
continuing. Please see our [Troubleshooting](help/troubleshooting.md optional=True) guide, which also contains
```

This needs to be optional to allow sockeye documentation to cleanly build (along with some other [changes I have pushed up to a PR there](https://github.inl.gov/ncrc/sockeye/pull/635)).

milljm replied:
Gonna to try and convince you all that using an https link, instead of internal... to get around this crazy rabbit hole.

cticenhour replied:
I would not be against that idea....it's just fragile if we re-organize things and forget to change it.

Commit:
Changes to help support other App documentation

Disconnect the Troubleshooting page. That site is a rabbit hole.

Separate the applications page a bit, so other apps can include
pieces of it, without displaying other apps.

cticenhour commented:
@milljm contents from the libMesh update PR made it into this along with other things....messy rebase?

milljm commented:
Yeah I really love git sometimes. I'll try to correct this.

Commit:
Forgot to include new files

Commit:
Do not isolate conda markdown

While I thought it was a good idea, no. Just change it back.

Commit:
Ugggggh, move this one file

...back to its correct location.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1325588/) on 8c1b06a wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>7549f6</th>
      <th colspan=4>#22922 8c1b06</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>84.62%</td>
      <td>84.62%</td>
      <td>+0.00%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>81976</td>
      <td>81980</td>
      <td>+4</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>14900</td>
      <td>14896</td>
      <td>-4</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/22922/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/22922/coverage/framework)


## Modules coverage

Inconsistent report tags were found between the head and base reports.
This can happen when reports are missing from either the head or the base.

Inconsistent tags:
  module-chemical_reactions
  module-chemical_reactions-parallel
  module-chemical_reactions-recover
  module-combined
  module-contact
  module-contact-parallel
  module-electromagnetics
  module-electromagnetics-parallel
  module-electromagnetics-recover
  module-external_petsc_solver
  module-external_petsc_solver-parallel
  module-external_petsc_solver-recover
  module-fluid_properties
  module-fluid_properties-parallel
  module-fluid_properties-recover
  module-fsi
  module-fsi-parallel
  module-fsi-recover
  module-functional_expansion_tools
  module-functional_expansion_tools-parallel
  module-functional_expansion_tools-recover
  module-geochemistry
  module-geochemistry-parallel
  module-heat_conduction
  module-heat_conduction-parallel
  module-heat_conduction-recover
  module-level_set
  module-level_set-parallel
  module-misc
  module-misc-parallel
  module-misc-recover
  module-navier_stokes
  module-navier_stokes-parallel
  module-navier_stokes-recover
  module-optimization
  module-optimization-parallel
  module-optimization-recover
  module-peridynamics
  module-peridynamics-parallel
  module-phase_field
  module-phase_field-parallel
  module-porous_flow
  module-porous_flow-parallel
  module-ray_tracing
  module-ray_tracing-parallel
  module-ray_tracing-recover
  module-rdg
  module-rdg-parallel
  module-reactor
  module-reactor-parallel
  module-richards
  module-scalar_transport
  module-scalar_transport-parallel
  module-scalar_transport-recover
  module-solid_properties
  module-solid_properties-parallel
  module-solid_properties-recover
  module-stochastic_tools
  module-stochastic_tools-parallel
  module-tensor_mechanics
  module-tensor_mechanics-parallel
  module-thermal_hydraulics
  module-thermal_hydraulics-parallel
  module-thermal_hydraulics-recover
  module-xfem
  module-xfem-parallel

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/22922/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/22922/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/22922/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/22922/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/22922/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/22922/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/22922/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/22922/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/22922/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/22922/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/22922/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/22922/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/22922/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/22922/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/22922/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/22922/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/22922/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/22922/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/22922/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/22922/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/22922/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/22922/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/22922/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/22922/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/22922/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/22922/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/22922/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/22922/coverage/xfem)

</details>



This comment will be updated on new commits.

loganharbour commented:
BISON and Sockeye patches are ready and pass.
